src/jtag/drivers/remote_bitbang.c

```diff
@@ -188,16 +189,14 @@ static int remote_bitbang_sample(void)
 static bb_value_t remote_bitbang_read_sample(void)
 {
 	if (remote_bitbang_recv_buf_start == remote_bitbang_recv_buf_end) {
-		if (remote_bitbang_fill_buf() != ERROR_OK)
-			return ERROR_FAIL;
-	}
-	if (remote_bitbang_recv_buf_start != remote_bitbang_recv_buf_end) {
-		int c = remote_bitbang_recv_buf[remote_bitbang_recv_buf_start];
-		remote_bitbang_recv_buf_start =
-			(remote_bitbang_recv_buf_start + 1) % sizeof(remote_bitbang_recv_buf);
-		return char_to_int(c);
+		if (remote_bitbang_fill_buf(BLOCK) != ERROR_OK)
+			return BB_ERROR;
 	}
-	return remote_bitbang_rread();
+	assert(remote_bitbang_recv_buf_start != remote_bitbang_recv_buf_end);
+	int c = remote_bitbang_recv_buf[remote_bitbang_recv_buf_start];
+	remote_bitbang_recv_buf_start =
+		(remote_bitbang_recv_buf_start + 1) % sizeof(remote_bitbang_recv_buf);
```

Comment:
Another snippet that should go to its own function - `remote_bitbang_recv_buf_pop_char()` or similar.